File: eBarbershop.Services/Database/Usluga.cs
using System;
using System.Collections.Generic;

namespace eBarbershop.Services.Database;

public  class Usluga
{
    public int UslugaId { get; set; }

    public string Naziv { get; set; } = null!;

    public string Opis { get; set; } = null!;

    public decimal Cijena { get; set; }

    public virtual ICollection<Rezervacija> Rezervacijas { get; set; } = new List<Rezervacija>();
}

File: UI/ebarbershop_mobile/lib/models/usluga.dart
import 'package:ebarbershop_mobile/models/rezervacija.dart';
import 'package:json_annotation/json_annotation.dart';

part 'usluga.g.dart';

@JsonSerializable()
class Usluga {
  int? uslugaId;
  String? naziv;
  String? opis;
  double? cijena;
  List<Rezervacija>? rezervacije;
  Usluga(this.uslugaId, this.naziv, this.opis, this.cijena, this.rezervacije);

  factory Usluga.fromJson(Map<String, dynamic> json) => _$UslugaFromJson(json);
  Map<String, dynamic> toJson() => _$UslugaToJson(this);
}
